src/plugins/legacy/polygonRoi/writers/medContoursWriters.cpp

```diff
@@ -0,0 +1,252 @@
+        if (file.open(QFile::WriteOnly))
+        {
+            QDataStream out(&file);
+
+            qint64 version = EXPORT_VERSION_1;
+            qint64 meshPos = 0;
+            qint64 contoursPos = 0;
+            qint64 endFilePos = 0;
+
+            out << version << meshPos << contoursPos << endFilePos;
+            meshPos = file.pos();
+
+            addMetaDataAsFieldData(mesh);
+
+            vtkSmartPointer<vtkDataWriter> writer = nullptr;
+            writer = vtkSmartPointer<vtkPolyDataWriter>::New();
+            writer->SetInputData(dynamic_cast<vtkPolyData*>(mesh->GetDataSet()));
+            writer->SetFileTypeToBinary();
+            writer->SetWriteToOutputString(1);
+            writer->SetHeader(qPrintable(header));
+            writer->Write();
+
+            // write to output file
+            int writeStatus = out.writeRawData(reinterpret_cast<const char*>(writer->GetBinaryOutputString()), writer->GetOutputStringLength());
+            if (writeStatus == -1)
+            {
+                qDebug() << metaObject()->className() << ":: write - unable to write mesh as binary data.";
+                file.close();
+                return false;
+            }
+
+            clearMetaDataFieldData(mesh);
+            contoursPos = file.pos();
+
+            QVector<medTagContours> &contours = ctr->getTagContoursSet();
+            out<<contours.size();
+            for (medTagContours& ctr : contours)
+                out<<ctr;
+            endFilePos = file.pos();
+
+            // rewrite file header
+            file.seek(0);
+            out << version << meshPos << contoursPos << endFilePos;
+
+            file.close();
+        }
+    }
+    catch(...)
+    {
+        qDebug()<< metaObject()->className() << "::write - error writing to "<<path;
+        clearMetaDataFieldData(mesh);
+        return false;
+    }
+    return true;
+}
+
+bool medContoursWriter::writeASCIIFile(medContours* ctr, const QString& path, vtkMetaDataSet* mesh)
+{
+    try
+    {
+        setlocale(LC_NUMERIC, "C");
+        QLocale::setDefault(QLocale("C"));
+        addMetaDataAsFieldData(mesh);
+        mesh->Write(path.toLocal8Bit().constData());
+        clearMetaDataFieldData(mesh);
+        QFile file(path);
+
+        if (file.open(QFile::Append))
+        {
+            QTextStream out(&file);
+            QVector<medTagContours> contoursSet = ctr->getTagContoursSet();
+            out<<"number of labels\n"<<contoursSet.size();
+            for (medTagContours& contours : contoursSet)
+            {
+                out<<contours;
+            }
+            file.close();
+        }
+    }
+    catch(...)
+    {
+        qDebug()<< metaObject()->className() << "::write - error writing to "<<path;
+        clearMetaDataFieldData(mesh);
+        return false;
+    }
+    return true;
+
+}
+
+void medContoursWriter::addMetaDataAsFieldData(vtkMetaDataSet* dataSet)
+{
+    foreach (QString key, data()->metaDataList())
```

Comment:
foreach (QString key, data()->metaDataList())
->
for (QString & key : data()->metaDataList())

Reply:
done